tests/unit/services/apigateway/test_handler_integration_response.py

```diff
@@ -93,3 +105,202 @@ def select_int_response(selection_value: str) -> IntegrationResponse:
+    context.region = TEST_AWS_REGION_NAME
+    context.account_id = TEST_AWS_ACCOUNT_ID
+    context.stage = TEST_API_STAGE
+    context.api_id = TEST_API_ID
+
+    request = InvocationRequestParser().create_invocation_request(context)
+    context.invocation_request = request
+
+    # add path_parameters from the router parser
+    # request["path_parameters"] = {"proxy": "path"}
+
+    context.resource_method = Method(
+        methodIntegration=Integration(type=IntegrationType.HTTP), methodResponses={}
+    )
+    context.context_variables = ContextVariables(
+        # resourceId="resource-id",
+        # apiId=TEST_API_ID,
+        # httpMethod="POST",
+        # path=f"{TEST_API_STAGE}/resource/{{proxy}}",
+        # resourcePath="/resource/{proxy}",
+        # stage=TEST_API_STAGE,
+    )
+    return context
+
+
+@pytest.fixture
+def create_default_response():
+    def _response():
+        return Response(
+            status=200,
+            headers={"content-type": "application/json", "header": ["multi", "header"]},
+            response=b'{"foo":"bar"}',
+        )
+
+    return _response
+
+
+@pytest.fixture
+def integration_response_handler():
+    """Returns a dummy integration response handler invoker for testing."""
+
+    def _handler_invoker(context: RestApiInvocationContext, response: Response):
+        return IntegrationResponseHandler()(RestApiGatewayHandlerChain(), context, response)
+
+    return _handler_invoker
+
+
+class TestHandlerIntegrationResponse:
+    def test_proxy_passthrough(
+        self, default_context, create_default_response, integration_response_handler
+    ):
+        og_response = create_default_response()
+        response = create_default_response()
+        default_context.resource_method["methodIntegration"]["type"] = IntegrationType.AWS_PROXY
+        integration_response_handler(default_context, response)
+        assert response.response == og_response.response
+        assert response.status_code == og_response.status_code
+        assert response.headers == og_response.headers
+
+        response = create_default_response()
+        default_context.resource_method["methodIntegration"]["type"] = IntegrationType.AWS_PROXY
+        integration_response_handler(default_context, response)
+        assert response.response == og_response.response
+        assert response.status_code == og_response.status_code
+        assert response.headers == og_response.headers
+
+    def test_status_code(
+        self, default_context, create_default_response, integration_response_handler
+    ):
+        integration_response = IntegrationResponse(
+            statusCode="300",
+            selectionPattern="",
+            responseParameters=None,
+            responseTemplates=None,
+        )
+        default_context.resource_method["methodIntegration"]["integrationResponses"] = {
+            "default": integration_response
```

Comment:
nit: I got a bit confused by the `default` naming here 😅 is it supposed to be status code of the integration response? not very important as we don't rely on it

Reply:
I actually meant to put `$default`, as to test the behaviour different with integration request, which would prioritize it.